### PR TITLE
Update API reference doc for core

### PR DESCRIPTION
**Pull request by jbogarthyde**
Add introductions to usage examples and edit descriptions to be more complete and consistent with current API reference styles

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Many examples have not descriptions or introduction. Language is not always clear or consistent.

Issue Number: N/A

## What is the new behavior?

Updates API doc in the core package for consistency of style. Mostly adds introduction to examples, 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/change_detection/change_detector_ref.ts [hunk 1]
```diff
@@ -9,10 +9,13 @@
 import {injectChangeDetectorRef as render3InjectChangeDetectorRef} from '../render3/view_engine_compatibility';
 
 /**
- * Base class for Angular Views, provides change detection functionality.
+ * Base class for `ViewRef` that provides change detection functionality.
```
Comment (pkozlowski-opensource) [inline packages/core/src/change_detection/change_detector_ref.ts:12]: The fact that `ViewRef` extends `ChangeDetectorRef` is the internal implementation detail and I don't think that we need to expose this information to the end users. How about "Class that provides change detection functionality"?

File: packages/core/src/linker/view_ref.ts [hunk 2]
```diff
@@ -10,12 +10,12 @@ import {ApplicationRef} from '../application_ref';
...
 
 /**
- * Represents an Angular [view](guide/glossary#view),
- * specifically the [host view](guide/glossary#view-tree) that is defined by a component.
- * Also serves as the base class
- * that adds destroy methods for [embedded views](guide/glossary#view-tree).
+ * Represents an Angular [view](guide/glossary#view "Definition"),
+ * specifically the [host view](guide/glossary#view-tree "Definition") that is defined by a
+ * component. Also serves as the base class that adds destroy methods for [embedded
+ * views](guide/glossary#view-tree "Definition").
```
Comment (pkozlowski-opensource) [inline packages/core/src/linker/view_ref.ts:16]: I know that this is a pre-existing text but I find it incorrect. In reality a `ViewRef` can represent any type of view (a component host view or an embedded view) while the text makes it sounds like it is mostly component view. 

How about simply saying "Represents an Angular view", link to the glossary and skip the rest?

File: packages/core/src/metadata/directives.ts [hunk 3]
```diff
@@ -314,7 +314,7 @@ export interface ComponentDecorator {
    * An Angular app contains a tree of Angular components.
    *
    * Angular components are a subset of directives, always associated with a template.
-   * Unlike other directives, only one component can be instantiated per an element in a template.
+   * Unlike other directives, only one component can be instantiated for one element in a template.
```
Comment (pkozlowski-opensource) [inline packages/core/src/metadata/directives.ts:317]: nit: my personal preference would go to the oryginal phrasing as for me it was better conveying the fact that there can be only one component for a given element. Maybe let's use "only one component can be instantiated for a given element in a template" if we want to get rid of "per"?

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM with few comments that I would like to see addressed before merging (hence cleanup label)

**Comment by jbogarthyde** [comment]
Docs only, should not need G3

**Pull request closed without merging**